azure/types.go

```diff
@@ -126,7 +126,6 @@ func (vmss VMSS) HasModelChanges(other VMSS) bool {
 	equal := cmp.Equal(vmss.Image, other.Image) &&
 		cmp.Equal(vmss.Identity, other.Identity) &&
 		cmp.Equal(vmss.Zones, other.Zones) &&
-		cmp.Equal(vmss.Tags, other.Tags) &&
```

Comment:
it looks like there are some unit tests that could be updated to verify this doesn't trigger on tags?  This would be more for validation/documentation in the future.

We are covered by e2e tests in some capacity so could be follow up improvement later when we add better logging.